cmd/kubeadm/app/cmd/join_test.go

```diff
@@ -73,7 +73,6 @@ func TestNewValidJoin(t *testing.T) {
 		skipPreFlight         bool
 		cfgPath               string
 		configToWrite         string
-		featureGatesString    string
```

Comment:
Aren't you still missing a negative test?  

Pass in FeatureGates and test the failure condition.